syntax_checker.py

```diff
@@ -0,0 +1,52 @@
+import coloredlogs, logging
+import os
+import queue
+
+from pathlib import Path
+from subprocess import Popen, PIPE
+from threading import Thread
+
+# logging setup
+logger = logging.getLogger(__name__)
+coloredlogs.install(level='DEBUG', logger=logger, isatty=True, fmt='%(levelname)s %(message)s')
+
+
+class SyntaxChecker:
+    command = ""
+    file_pattern = ""
+    def __init__(self, thread_count=5):
+        self.thread_count = thread_count
+        self.results = []
+        self.workpool = queue.Queue()
+
+    def lint_files(self):
+        while not self.workpool.empty():
+            filename = self.workpool.get()
+
+            command = [self.command, filename]
+
+            p = Popen(command, stdout=PIPE, stderr=PIPE)
+            stdout, stderr = p.communicate()
+
+            for result in self.get_results(stdout, stderr):
+                self.results.append(result)
+
+
+    def get_results(self, stdout, stderr):
+        raise NotImplementedEror
```

Comment:
What's your thinking with this?

Reply:
the only places where all the build scripts really differ is when getting results from stdout/stderr
so I put that in its own function and it cant really be defined in the base class